pkg/metrics/metrics.jsx

```diff
@@ -873,46 +874,114 @@ class MetricsHour extends React.Component {
 }
 
 const persistentServiceState = proxy => ['running', 'stopped', 'failed', undefined].indexOf(proxy.state) >= 0;
+const validServiceState = proxy => ['running', 'stopped'].indexOf(proxy.state) >= 0;
 
-const PCPConfig = ({ buttonVariant }) => {
+const PCPConfig = ({ buttonVariant, firewalldRequest }) => {
     const [dialogVisible, setDialogVisible] = useState(false);
     const [dialogError, setDialogError] = useState(null);
     const [dialogLoggerValue, setDialogLoggerValue] = useState(false);
+    const [dialogProxyValue, setDialogProxyValue] = useState(null);
+    const [dialogInitialProxyValue, setDialogInitialProxyValue] = useState(null);
     const [pending, setPending] = useState(false);
-    const s_pmlogger = useObject(() => service.proxy("pmlogger.service"), null, []);
 
-    debug("PCPConfig s_pmlogger.state", s_pmlogger.state);
+    const s_pmlogger = useObject(() => service.proxy("pmlogger.service"), null, []);
+    const s_pmproxy = useObject(() => service.proxy("pmproxy.service"), null, []);
+    // redis.service on Fedora/RHEL, redis-server.service on Debian/Ubuntu with an Alias=redis
+    const s_redis = useObject(() => service.proxy("redis.service"), null, []);
+    const s_redis_server = useObject(() => service.proxy("redis-server.service"), null, []);
 
-    useEvent(s_pmlogger, "changed");
     useEvent(superuser, "changed");
+    useEvent(s_pmlogger, "changed");
+    useEvent(s_redis, "changed");
+    useEvent(s_redis_server, "changed");
+
+    let real_redis;
+    let redis_name;
+    if (s_redis_server.exists) {
+        real_redis = s_redis_server;
+        redis_name = "redis-server.service";
+    } else {
+        real_redis = s_redis;
+        redis_name = "redis.service";
+    }
+
+    debug("PCPConfig s_pmlogger.state", s_pmlogger.state);
+    debug("PCPConfig s_pmproxy state", s_pmproxy.state, "redis exists", s_redis.exists, "state", s_redis.state, "redis-server exists", s_redis_server.exists, "state", s_redis_server.state);
 
     if (!superuser.allowed)
         return null;
 
     const handleSave = () => {
```

Comment:
Ideal would be to use isLoading Button property while the dialog is processing the save. And of course disable it. Because it takes usually a few seconds till the dialog is closed.

Reply:
Disabling should already happen through `isDisabled={pending}`. But good point about the `isLoading`, I'll try that.

Reply:
Done, in a separate commit as that dialog already exist in master. Thanks for the suggestion!